File: app/src/main/java/info/firozansari/animatedvectordrawable/AnimatedIconActivity.java
package info.firozansari.animatedvectordrawable;

import android.os.Bundle;
import android.support.graphics.drawable.AnimatedVectorDrawableCompat;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;

import java.util.ArrayList;

/**
 * Created by firoz on 19/03/2017.
 */

public class AnimatedIconActivity extends AppCompatActivity {
    private ArrayList<AnimatedIcon> animatedIcons = new ArrayList<>();
    private IconAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.animated_icon);

        // Init and Setup RecyclerView
        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.list);
        recyclerView.setLayoutManager(new GridLayoutManager(AnimatedIconActivity.this, 2));
        //animator not yet working
        recyclerView.setItemAnimator(new DefaultItemAnimator());

        mAdapter = new IconAdapter();
        recyclerView.setAdapter(mAdapter);

    }

    @Override
    protected void onResume() {
        super.onResume();

        AnimatedVectorDrawableCompat avd = AnimatedVectorDrawableCompat.create(AnimatedIconActivity.this, R.drawable.anim_android);
        AnimatedIcon icon1 = new AnimatedIcon(avd, getString(R.string.android_icon));
        animatedIcons.add(icon1);

        mAdapter.setAnimatedIcons(animatedIcons);


    }
}

File: app/src/main/java/info/firozansari/animatedvectordrawable/AnimatedIconActivity.kt
package info.firozansari.animatedvectordrawable

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import androidx.vectordrawable.graphics.drawable.AnimatedVectorDrawableCompat
import java.util.ArrayList

/**
 * Created by firoz on 19/03/2017.
 */
class AnimatedIconActivity : AppCompatActivity() {
    private val animatedIcons = ArrayList<AnimatedIcon>()
    private var mAdapter: IconAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.animated_icon)

        // Init and Setup RecyclerView
        val recyclerView: RecyclerView = findViewById<RecyclerView>(R.id.list)
        recyclerView.layoutManager = GridLayoutManager(this@AnimatedIconActivity, 2)
        //animator not yet working
        recyclerView.itemAnimator = DefaultItemAnimator()
        mAdapter = IconAdapter()
        //recyclerView.setAdapter(mAdapter)
    }

    override fun onResume() {
        super.onResume()
        val avd1: AnimatedVectorDrawableCompat? =
            AnimatedVectorDrawableCompat.create(this@AnimatedIconActivity, R.drawable.anim_android)
        val avd2: AnimatedVectorDrawableCompat? =
            AnimatedVectorDrawableCompat.create(this@AnimatedIconActivity, R.drawable.anim_blutooth)
        val icon1 = avd1?.let { AnimatedIcon(it, getString(R.string.android_icon)) }
        val icon2 = avd2?.let { AnimatedIcon(it, getString(R.string.bluetooth_icon)) }
        icon1?.let { animatedIcons.add(it) }
        icon2?.let { animatedIcons.add(it) }
        mAdapter?.setAnimatedIcons(animatedIcons)
    }
}
